logstash-core/src/test/java/org/logstash/stress/Concurrent.java

```diff
@@ -17,32 +9,36 @@
 import java.util.List;
 import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.stream.Collectors;
+import org.logstash.ackedqueue.Batch;
+import org.logstash.ackedqueue.SettingsImpl;
+import org.logstash.ackedqueue.Queue;
+import org.logstash.ackedqueue.Settings;
+import org.logstash.ackedqueue.StringElement;
+import org.logstash.ackedqueue.io.ByteBufferPageIO;
+import org.logstash.ackedqueue.io.CheckpointIOFactory;
+import org.logstash.ackedqueue.io.FileCheckpointIO;
+import org.logstash.ackedqueue.io.MemoryCheckpointIO;
+import org.logstash.ackedqueue.io.MmapPageIO;
+import org.logstash.ackedqueue.io.PageIOFactory;
 
 public class Concurrent {
     final static int ELEMENT_COUNT = 2000000;
     final static int BATCH_SIZE = 1000;
     static Settings settings;
 
     public static Settings memorySettings(int capacity) {
-        Settings s = new MemorySettings();
         PageIOFactory pageIOFactory = (pageNum, size, path) -> new ByteBufferPageIO(pageNum, size, path);
         CheckpointIOFactory checkpointIOFactory = (source) -> new MemoryCheckpointIO(source);
-        s.setCapacity(capacity);
-        s.setElementIOFactory(pageIOFactory);
-        s.setCheckpointIOFactory(checkpointIOFactory);
-        s.setElementClass(StringElement.class);
-        return s;
+        return SettingsImpl.memorySettingsBuilder().capacity(capacity).elementIOFactory(pageIOFactory)
+            .checkpointIOFactory(checkpointIOFactory).elementClass(StringElement.class).build();
     }
 
     public static Settings fileSettings(int capacity) {
-        Settings s = new MemorySettings("/tmp/queue");
         PageIOFactory pageIOFactory = (pageNum, size, path) -> new MmapPageIO(pageNum, size, path);
         CheckpointIOFactory checkpointIOFactory = (source) -> new FileCheckpointIO(source);
-        s.setCapacity(capacity);
-        s.setElementIOFactory(pageIOFactory);
-        s.setCheckpointIOFactory(checkpointIOFactory);
-        s.setElementClass(StringElement.class);
-        return s;
+        return SettingsImpl.memorySettingsBuilder("/tmp/queue").capacity(capacity)
```

Comment:
@robbavey urgh `"/tmp/queue"` hard set is evil ... we gotta add an issue for that